sirepo/package_data/static/js/raydata.js

```diff
@@ -616,6 +616,61 @@ SIREPO.app.directive('pngImage', function(plotting) {
     };
 });
 
+SIREPO.app.directive('elementPicker', function() {
+    return {
+        restrict: 'A',
+        scope: {
+            controller: '=',
+            title: '@',
+            id: '@',
+            smallElementClass: '@',
+            availableColumns: '=',
+            masterListColumns: '=',
+            columns: '=',
+        },
+        template: [
+            '<div class="modal fade" data-ng-attr-id="{{ id }}" tabindex="-1" role="dialog">',
+              '<div class="modal-dialog modal-lg">',
+                '<div class="modal-content">',
+                  '<div class="modal-header bg-info">',
+                    '<button type="button" class="close" data-dismiss="modal"><span>&times;</span></button>',
+                    '<span class="lead modal-title text-info">{{ title }}</span>',
+                  '</div>',
+                  '<div class="modal-body">',
+                    '<div class="container-fluid">',
+                      '<label for="scans-columns">Field: </label>',
+                        '<select name="scans-columns" id="scans-columns" data-ng-model="selected" ng-change="selectColumn()">',
+                          '<option ng-repeat="column in availableColumns">{{column}}</option>',
+                        '</select>',
+                    '</div>',
+                  '</div>',
+                '</div>',
+              '</div>',
+            '</div>',
+        ].join(''),
+        controller: function($scope, appState, $interval, panelState) {
+            $scope.activeTab = 'basic';
+            $scope.selected = null;
+            $scope.columns = ['select'].concat(appState.models.metadataColumns);
+
+            $scope.selectColumn = function() {
+                if ($scope.selected !== null) {
+                    appState.models.metadataColumns.push($scope.selected);
+                }
+                appState.saveChanges('metadataColumns');
+                $scope.columns = ['select'].concat(appState.models.metadataColumns);
+
+                $scope.selected = null;
+                $scope.availableColumns = $scope.masterListColumns.filter(function(value) {
```

Comment:
This code is repeated in scanSelector. Can it be removed here or there?

Reply:
07e13b7b7da132d5450c7714b0bcd7a04982308c